packages/apps/spaces/app/organizations/[id]/components/Timeline/events/email/EmailStub.tsx

```diff
@@ -5,12 +5,15 @@ import { Text } from '@ui/typography/Text';
 import { VStack } from '@ui/layout/Stack';
 import { convert } from 'html-to-text';
 import { getEmailParticipantsName } from '@spaces/utils/getParticipantsName';
-import { EmailParticipant, InteractionEvent } from '@graphql/types';
+import { EmailParticipant } from '@graphql/types';
 import { useTimelineEventPreviewContext } from '@organization/components/Timeline/preview/TimelineEventsPreviewContext/TimelineEventPreviewContext';
 import { getEmailParticipantsByType } from '@organization/components/Timeline/events/email/utils';
 import Image from 'next/image';
+import { InteractionEventWithDate } from '../../types';
 
-export const EmailStub: FC<{ email: InteractionEvent }> = ({ email }) => {
+export const EmailStub: FC<{ email: InteractionEventWithDate }> = ({
```

Comment:
In the new react version (which we use) FC does not include optional children props so it does not geopardise type safety
Using the React.FC<Props> approach provides a clear indication that a function is a functional React component. It also opens up the possibility for utilizing static properties such as defaultProps, propTypes, and contextTypes, which can further enhance component with default properties, enforcing property type checking, or using context within it.

Both approaches to type components are valid and it often comes down to personal preference, in my opinion if we want to enforce single standard I would propose to use FC for reasons mentioned above

Reply:
propTypes is useless in context of typescript, defaultProps maybe can add some value although it's easier to just do `{ someProps = 'default' }` and adding default to properties is always better to be done from a data mapping layer as for contextTypes I don't know what to say... seems very abstract and low level and never ever used it. Nonetheless, my opinion about this is that we optimise ahead of time for something that didn't yet occurred. We already now that a react component must start with an `UpperCase` letter in the same manner we know a hook must start with `use`. A while ago the whole react community [advised](https://github.com/facebook/create-react-app/pull/8177) against React.FC and everybody started moving away from it. For me it was making sense when PropsWithChildren was not yet added to react typing and you always had to manually type the children and even then, people would make components with FC even though the children prop was never used. I've been though this once with FC and I really don't want to start refactoring away from it because it becomes infamous again.